var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -41,6 +41,8 @@ class Protobuf(AutotoolsPackage):
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
 
+    conflicts('%gcc@:4.8') # Requires c++11
```

Comment:
Don't GCC 4.7 and 4.8 support C++11?
https://gcc.gnu.org/projects/cxx-status.html